src/hmc.py

```diff
@@ -0,0 +1,298 @@
+                        'HMC_SENT_R_LOW_LINK2': 0x28,
+                        'HMC_SENT_R_HIGH_LINK2': 0x2C,
+                        'HMC_POISONED_PACKET_LOW_LINK2': 0x30,
+                        'HMC_POISONED_PACKET_HIGH_LINK2': 0x34,
+                        'HMC_RCVD_RESP_LOW_LINK2': 0x38,
+                        'HMC_RCVD_RESP_HIGH_LINK2': 0x3C,
+                        'HMC_TX_LINK_RETRIES_LOW_LINK2': 0x40,
+                        'HMC_TX_LINK_RETRIES_HIGH_LINK2': 0x44,
+                        'HMC_ERR_ON_RX_LOW_LINK2': 0x48,
+                        'HMC_ERR_ON_RX_HIGH_LINK2': 0x4C,
+                        'HMC_RUN_LNGTH_BITFLIP_LOW_LINK2': 0x50,
+                        'HMC_RUN_LNGTH_BITFLIP_HIGH_LINK2': 0x54,
+                        'HMC_ERR_ABORT_NOT_CLEAR_LOW_LINK2': 0x58,
+                        'HMC_ERR_ABORT_NOT_CLEAR_HIGH_LINK2': 0x5C,
+                        'HMC_ERR_RSP_PACKET_LINK2': 0x60,
+                        'HMC_ERRSTAT_LINK2': 0x64,
+                        'HMC_CRC_ERR_CNT_LINK2': 0x68,
+                        # HMC Status Registers (LINK3)
+                        'HMC_STAT_GEN_LOW_LINK3': 0x6C,
+                        'HMC_STAT_GEN_HIGH_LINK3': 0x70,
+                        'HMC_STAT_INIT_LOW_LINK3': 0x74,
+                        'HMC_STAT_INIT_HIGH_LINK3': 0x78,
+                        'HMC_CTRL_LOW_LINK3': 0x7C,
+                        'HMC_CTRL_HIGH_LINK3': 0x80,
+                        'HMC_SENT_P_LOW_LINK3': 0x84,
+                        'HMC_SENT_P_HIGH_LINK3': 0x88,
+                        'HMC_SENT_NP_LOW_LINK3': 0x8C,
+                        'HMC_SENT_NP_HIGH_LINK3': 0x90,
+                        'HMC_SENT_R_LOW_LINK3': 0x94,
+                        'HMC_SENT_R_HIGH_LINK3': 0x98,
+                        'HMC_POISONED_PACKET_LOW_LINK3': 0x9C,
+                        'HMC_POISONED_PACKET_HIGH_LINK3': 0xA0,
+                        'HMC_RCVD_RESP_LOW_LINK3': 0xA4,
+                        'HMC_RCVD_RESP_HIGH_LINK3': 0xA8,
+                        'HMC_TX_LINK_RETRIES_LOW_LINK3': 0xAC,
+                        'HMC_TX_LINK_RETRIES_HIGH_LINK3': 0xB0,
+                        'HMC_ERR_ON_RX_LOW_LINK3': 0xB4,
+                        'HMC_ERR_ON_RX_HIGH_LINK3': 0xB8,
+                        'HMC_RUN_LNGTH_BITFLIP_LOW_LINK3': 0xBC,
+                        'HMC_RUN_LNGTH_BITFLIP_HIGH_LINK3': 0xC0,
+                        'HMC_ERR_ABORT_NOT_CLEAR_LOW_LINK3': 0xC4,
+                        'HMC_ERR_ABORT_NOT_CLEAR_HIGH_LINK3': 0xC8,
+                        'HMC_ERR_RSP_PACKET_LINK3': 0xCC,
+                        'HMC_ERRSTAT_LINK3': 0xD0,
+                        'HMC_CRC_ERR_CNT_LINK3': 0xD4,
+                        'HMC_STATUS': 0xD8}
+        # dictionary holding all HMC status information
+        self.hmc_status_list = {}
+        #dictionary holding all HMC revision information
+        self.hmc_revision_list = {}
+        LOGGER.debug('New Hmc %s' % self.__str__())
```

Comment:
Print the device_name instead of the __str__ representation, so the user knows which HMC this is.

Reply:
Agreed. Fixed.